src/transformers/integrations/deepspeed.py

```diff
@@ -287,6 +288,14 @@ def is_deepspeed_zero3_enabled():
         return _hf_deepspeed_config_weak_ref().is_zero3()
     else:
         return False
+    
+def is_deepspeed_zero3_enabled_accelerate():
```

Comment:
Is it possible that this can evaluate to `True` and `is_deepspeed_zero3_enabled` to `False`?

Reply:
Yes, if the deepspeed config ref hasn't been made yet, which is why the check is needed. 

This exact situation comes from that :)

Reply:
I've expanded the docstring for a clearer description